Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx

```diff
@@ -136,9 +159,47 @@ export default function RegistrationAdministrationList() {
   const { isLoading: infoLoading, data: registrationsWithUser } =
     useWithUserData(registrations ?? [])
 
+  const sortedRegistrationWithUser = useMemo(() => {
+    if (registrationsWithUser) {
+      const sorted = registrationsWithUser.sort((a, b) => {
+        switch (sortColumn) {
+          case 'name':
+            return a.user.name.localeCompare(b.user.name)
+          case 'wca_id':
+            return a.user.wca_id.localeCompare(b.user.wca_id)
+          case 'country':
+            return a.user.country.name.localeCompare(b.user.country.name)
+          case 'events':
+            return a.competing.event_ids.length - b.competing.event_ids.length
+          case 'guests':
+            return a.guests - b.guests
+          case 'dob':
+            return a.user.dob - b.user.dob
+          case 'comment':
+            return a.competing.comment.localeCompare(b.competing.comment)
+          case 'registered_on':
+            return a.competing.registered_on.localeCompare(
+              b.competing.registered_on,
+            )
+          case 'paid_on':
+            return a.payment.updated_at.localeCompare(b.payment.updated_at)
+          default:
+            return 0
+        }
+      })
+      if (sortDirection === 'descending') {
+        return sorted.reverse()
+      }
+      return sorted
+    }
+    return []
+  }, [registrationsWithUser, sortColumn, sortDirection])
+
   const { waiting, accepted, cancelled, pending } = useMemo(
-    () => partitionRegistrations(registrationsWithUser ?? []),
-    [registrationsWithUser],
+    () => partitionRegistrations(sortedRegistrationWithUser ?? []),
+    // this useMemo will not fire again if we only change the order of the array, which is why we need the sortColumn and sortDirection in the deps
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [sortedRegistrationWithUser, sortColumn, sortDirection],
```

Comment:
I don't get this. `sortedRegistrationWithUser` is itself a `useMemo`, whose dependencies contain `sortColumn` and `sortDirection`.

You are arguing that this current memo needs to trigger if sort order changes, but when that happens the memo of `sortedRegistrationsWithUser` triggers, and by transitive rule this memo should then also trigger. So from what I can tell by staring at the code, your extra deps are totally unnecessary.

There has to be a cleaner solution than overriding good React principles and standards.

Reply:
Wow, thanks for explaining, that would probably have taken me half an hour to figure out :smile: 
I'm tempted to start a rant why on earth JS sorts in-place, but I guess the short answer is "legacy reasons"...

Made a PR over at https://github.com/thewca/wca-registration/pull/490

Reply:
Half an hour wouldn't be so bad; it took me several months to understand why my code didn't work as expected when I first started React...